src/libstore/build.cc

```diff
@@ -1629,6 +1629,19 @@ void DerivationGoal::buildDone()
            being valid. */
         registerOutputs();
 
+        if (settings.postBuildHook != "") {
+            printMsg(lvlChatty, format("executing post-build hook '%1%'")
+                % settings.postBuildHook);
+            auto outputPaths = drv->outputPaths();
+            std::map<std::string, std::string> hookEnvironment = getEnv();
+            hookEnvironment.emplace("DRV_PATH", drvPath);
+            hookEnvironment.emplace("OUT_PATHS", chomp(concatStringsSep(":", outputPaths)));
```

Comment:
I am thinking maybe a middle ground would be better: all arguments to the program are output paths, every other parameter passed over the environment. What do you think @edolstra? That way, truly nothing would need to be parsed.

Reply:
It seems more complicated to explain if the parameters are passed both in the env and the args.